### PR TITLE
RYETH/BTC allow Aave V3 adaptor for RYETH/BTC, Cellar adaptor for RYBTC, cachePriceRouter function

### PR DESCRIPTION
**Pull request by cbrit**
Note the cellar v2.2 ABI file changed, but the only strategist function affected was `cachePriceRouter` which is why it was necessary. Somehow RYETH was on a newer version of the ABI than we had in the repo.

### CODE REVIEW

File: proto/cellar_v2.proto [hunk 1]
```diff
@@ -387,6 +389,20 @@ message CellarV2_2 {
     message RemovePositionFromCatalogue {
         uint32 position_id = 1;
     }
+
+    /*
+    * Updates the cellar to use the latest price router in the registry. 
+    *
+    * Represents function `cachePriceRouter(bool checkTotalAssets, uint16 allowableRange, address, expectedPriceRouter)`
```
Comment (EricBolten) [inline proto/cellar_v2.proto]: Nit: `address, ` -> `address `

File: steward/src/cellars.rs [hunk 2]
```diff
@@ -116,6 +129,8 @@ pub const ADAPTOR_COMPOUND_C_TOKEN_V1: &str = "26dba82495f6189dde7648ae88bead46c
 
 // adaptors
 
+pub const ADAPTOR_AAVE_V3_A_TOKEN_V1: &str = "76cef5606c8b6ba38fe2e3c639e1659afa530b47";
+pub const ADAPTOR_CELLAR_V1: &str = "1e22adf9e63ef8f2a3626841ddddd19683e31068";
```
Comment (EricBolten) [inline steward/src/cellars.rs:133]: Why are we using V1 instead of V2?

File: steward/src/cellars/cellar_v2_2.rs [hunk 3]
```diff
@@ -233,6 +233,20 @@ pub fn get_encoded_function(call: FunctionCall, cellar_id: String) -> Result<Vec
...
+        Function::CachePriceRouter(params) => {
+            validate_cache_price_router(&cellar_id)?;
+            log_cellar_call(
+                CELLAR_NAME,
+                &CachePriceRouterCall::function_name(),
+                &cellar_id,
+            );
+            let call = CachePriceRouterCall {
+                check_total_assets: params.check_total_assets,
+                allowable_range: params.allowable_range as u16,
```
Comment (EricBolten) [inline steward/src/cellars/cellar_v2_2.rs:249]: The proto message includes an expected address, is there a reason it's unset here?

File: steward/src/cellars.rs [hunk 4]
```diff
@@ -204,6 +219,15 @@ pub fn validate_oracle(
     Ok(())
 }
 
+pub fn validate_cache_price_router(cellar_id: &str) -> Result<(), Error> {
```
Comment (EricBolten) [inline steward/src/cellars.rs]: Sun mentioned we were expecting `true` and `500` for the args. Should we validate against that?

**Review by EricBolten — COMMENTED** [review]
Some questions and comments.

File: steward/src/cellars.rs (hunk 2 above)
Comment (cbrit) [reply steward/src/cellars.rs:133]: As noted [here](https://docs.google.com/spreadsheets/u/2/d/1Kf_DT0bWC1imvr5fVsgiYZAHQ1IsYqw8KEycwAgVI0E/edit#gid=0) this change is for the "legacy" cellar adaptor

File: steward/src/cellars/cellar_v2_2.rs (hunk 3 above)
Comment (cbrit) [reply steward/src/cellars/cellar_v2_2.rs:249]: Ah need to fix the proto. I made it match the 2.5 cellar abi by mistake.

File: steward/src/cellars.rs (hunk 4 above)
Comment (cbrit) [reply steward/src/cellars.rs]: I can, do we usually validate non-identity arguments?

Comment (EricBolten) [reply steward/src/cellars.rs]: I'm basing it off of their requests. I assume this is a one-off and not something that is meant to be called much in the future, but you could verify with them.

**Review by EricBolten — APPROVED** [review]
LGTM

**Pull request merged**